python/tests/aggregate_tests/aggtst_base.py

```diff
@@ -87,48 +92,25 @@ def as_sql_insert(self) -> str:
         return stat + ";"
 
 
-def consolidate(data: JSON) -> JSON:
-    """Attempt to consolidate some results that contains identical rows with +1/-1.
-       Not a complete solution, but solves a real problem for aggregate tests
-       where we receive 2 lists of None values that cancel each other"""
-    insert = list(filter(lambda x: x["insert_delete"] == 1, data))
-    delete = list(filter(lambda x: x["insert_delete"] == -1, data))
-    for x in delete:
-        x["insert_delete"] = 1
-    result = []
-    for x in insert:
-        if x in delete:
-            delete.remove(x)
-            continue
-        del x['insert_delete']
-        result.append(x)
-    if len(delete) != 0:
-        raise("Could not normalize data", data)
-    return result
-
 class View(SqlObject):
     """A SQL view with contents"""
+
     def __init__(self, sql: str, data: JSON):
         super().__init__(SqlObject.extract_name(sql, False), sql, data)
-        self.listener = None
-
-    def listen(self, pipeline: Pipeline):
-        """Listen to an output handler
-
-        arguments:
-        pipeline -- pipeline to listen to"""
-        self.listener = pipeline.listen(self.name)
 
-    def validate(self):
+    def validate(self, pipeline: Pipeline):
         """Check that the data received matches the expected data"""
-        data = self.listener.to_dict()
-        data = consolidate(data)
+        data = list(pipeline.query(f"SELECT * FROM {self.name};"))
         expected = self.get_data()
-        assert expected == data , f"ASSERTION ERROR: failed view:{self.name}: {expected} != {data}"
+        expected = [{k: v for k, v in d.items() if v is not None} for d in expected]
+
+        tc = unittest.TestCase()
+        tc.assertCountEqual(data, expected), f"ASSERTION ERROR: failed view:{self.name}: {expected} != {data}"
```

Comment:
won't `assertCountEqual` print left and right sides automatically? Do you really need the third argument here?

Reply:
We don't. 

Removed.